### PR TITLE
Allow empty values for frost-date-picker

### PR DESCRIPTION
**Pull request by gknoy**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

This PR re-adds the ability for this component to have an `undefined` value, previously implemented in [PR 32](https://github.com/ciena-frost/ember-frost-date-picker/pull/32). This capability is required to let us use date pickers in Bunsen forms which have _optional_ date values.

### Screenshot
screen shot 2017-09-12 at 9 32 03 pm

 # Changelog
- The `value` property of `frost-date-picker` is again optional, so that
  it will work correctly with Bunsen forms which require _optional_ date
  fields.
- Added demo styling and headings to clarify examples
- Added code samples to widget demo
- Pinned more dependencies to fix the build

### CODE REVIEW

**Comment by sophypal** [comment]
Looks good to me but I don't think you should be committing the `yarn.lock`.

**Comment by gknoy** [comment]
@juwara0 said something similar, in that `npm` 5 makes it unnecessary. However, our Travis builds don't use npm 5 (they use 3).  Why not add a `yarn.lock` now, so that builds are predictable and don't surprisingly fail based on the calendar, and then remove it later once we start using npm 5?

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,9461 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (juwara0) [inline yarn.lock]: This file needs to be removed as discussed.

**Comment by notmessenger** [comment]
@gknoy Because the Travis builds are not using `yarn` either so this file is useless.

**Comment by gknoy** [comment]
@notmessenger That's a good point; it looks like that's a configuration setting we could change easily, as `ember-try` looks for one and will use it if you want it to.

@juwara0  I'll remove it, but I'd like to ask why are we opposed to using `yarn`? The build problems that we were struggling with all yesterday would have been prevented had we had a build with predictable package installation, and saved many hours of developers' time.  **The build should not be broken when checking out master**.

Saying "A future version of a tool we're not yet using fixes this too ..." ignores the benefits that we can get currently.

**Comment by sophypal** [comment]
@gknoy I understand your point. I would love to have dependencies locked down but `npm@3` also has that ability with `npm-shrinkwrap`, albeit with problems with peer dependencies. Not everyone has agreed to use `yarn` and for that fact, committing the `yarn.lock` will result in that file getting out of sync causing problems for people using `yarn`.

**Comment by sophypal** [comment]
👍

**Comment by gknoy** [comment]
@sophypal Thank you for your explanation about shrinkwrap. :) Good points.

**Comment by gknoy** [comment]
Why does coveralls show no data about the coverage? (https://coveralls.io/builds/13256665) I'm having trouble finding some kind of log about why it's not finding anything. =/

**Pull request merged**